aiidalab/app.py

```diff
@@ -78,12 +78,23 @@ class _AiidaLabApp:
     metadata: dict[str, Any]
     name: str
     path: Path
+    # TODO: It would be nicer to use parsed packaging.Version as a key instead of str
+    # That way it could also be pre-sorted.
     releases: dict[str, Any] = field(default_factory=dict)
 
     @classmethod
     def from_registry_entry(
         cls, path: Path, registry_entry: dict[str, Any]
     ) -> _AiidaLabApp:
+        # Filter out invalid versions
+        if releases := registry_entry.get("releases"):
+            versions = list(releases.keys())
+            for version in versions:
+                if not is_valid_version(version):
```

Comment:
This is the actual fix for #339, we skip the invalid versions when we create the `_AiidaLabApp` class instance, so any further code that accesses `self.releases` can assume that the versions are valid.